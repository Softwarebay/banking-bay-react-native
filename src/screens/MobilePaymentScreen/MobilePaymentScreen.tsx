import { MobilePaymentTemplate } from 'components/templates';

export const MobilePaymentScreen = () => {
	const onSuccessMobilePayment = (mobileNumber: string, amount: number) => {
		console.log('onSuccessMobilePayment', mobileNumber, amount);
	};
	return (
		<MobilePaymentTemplate
			inputPlaceholder="+17 | xxxxxxxxxx"
			onSuccessMobilePayment={onSuccessMobilePayment}
		/>
	);
};
