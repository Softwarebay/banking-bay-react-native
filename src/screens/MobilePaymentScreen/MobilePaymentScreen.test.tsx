import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { MobilePaymentScreen } from './MobilePaymentScreen';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
	useNavigation: () => ({ navigate: mockNavigate }),
}));

describe('MobilePaymentScreen', () => {
	it('should match snapshot', () => {
		const tree = testRenderer.create(<MobilePaymentScreen />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should display '+17 | xxxxxxxxxx' as mobile number input placeholder", () => {
		render(<MobilePaymentScreen />);

		const mobileInputField = screen.getByTestId(
			'input-field-mobile-payment-input-test-id'
		);

		expect(mobileInputField.props).toHaveProperty(
			'placeholder',
			'+17 | xxxxxxxxxx'
		);
	});

	it("should log 'onSuccessMobilePayment', mobile number and amount after successful payment", () => {
		render(<MobilePaymentScreen />);
		console.log = jest.fn();

		const mobileInputField = screen.getByTestId(
			'input-field-mobile-payment-input-test-id'
		);
		const confirmBtn = screen.getByTestId('confirm-mobile-payment-btn-test-id');

		fireEvent.changeText(mobileInputField, '123456');
		fireEvent.press(confirmBtn);

		expect(console.log).toHaveBeenCalledWith(
			'onSuccessMobilePayment',
			'123456',
			10.0
		);
	});

	it('should navigate to SuccessPaymentScreen after successful payment', () => {
		render(<MobilePaymentScreen />);

		const mobileInputField = screen.getByTestId(
			'input-field-mobile-payment-input-test-id'
		);
		const confirmBtn = screen.getByTestId('confirm-mobile-payment-btn-test-id');

		fireEvent.changeText(mobileInputField, '123456');
		fireEvent.press(confirmBtn);

		expect(mockNavigate).toHaveBeenCalledWith('SuccessPayment');
	});
});
