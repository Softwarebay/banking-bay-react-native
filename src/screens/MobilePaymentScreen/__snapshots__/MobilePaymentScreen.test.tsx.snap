// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobilePaymentScreen should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FAFCFF",
      "flex": 1,
      "paddingBottom": 30,
      "paddingHorizontal": 20,
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "shadowColor": "#161966",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.08,
            "shadowRadius": 16,
          },
          Array [
            Object {
              "height": 55,
              "paddingHorizontal": 20,
              "paddingVertical": 14,
            },
            undefined,
          ],
        ],
      ]
    }
    testID="mobile-payment-input-test-id"
  >
    <SvgMock
      height={20}
      testID="flag-usa-icon-test-id"
      width={20}
    />
    <TextInput
      autoCorrect={false}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="+17 | xxxxxxxxxx"
      style={
        Array [
          Object {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 15,
          },
          Object {
            "flex": 1,
            "marginLeft": 8,
          },
        ]
      }
      testID="input-field-mobile-payment-input-test-id"
      value=""
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 17,
          "lineHeight": 27.2,
        },
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 22.4,
          },
          Object {
            "marginBottom": 10,
            "marginTop": 18,
          },
        ],
      ]
    }
    testID="mobile-payment-balance-text-test-id"
  >
    Your balance: 4863.27 USD
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Object {
              "marginRight": 15,
              "paddingHorizontal": 22,
              "paddingVertical": 18,
            },
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Array [
              Object {
                "alignItems": "baseline",
              },
              undefined,
            ],
          ]
        }
        testID="mobile-payment-float-num-test-id"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Array [
                Object {
                  "fontFamily": "MulishMedium",
                },
                Object {
                  "fontSize": 33,
                },
              ],
            ]
          }
        >
          $10.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Array [
                Object {
                  "fontFamily": "MulishMedium",
                },
                Object {
                  "fontSize": 20,
                },
              ],
            ]
          }
        >
          00
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          Object {
            "fontSize": 14,
            "lineHeight": 22.4,
          },
        ]
      }
      testID="mobile-payment-fees-text-test-id"
    >
      No fees
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Object {
          "marginTop": "auto",
        },
      ]
    }
    testID="confirm-mobile-payment-btn-test-id"
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4288142079,
            4279654301,
          ],
          "endPoint": Array [
            1,
            1,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            0,
          ],
        }
      }
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 55,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "MulishSemiBold",
            "fontSize": 17,
          }
        }
      >
        Confirm
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;
